src/frontends/src/bench.cpp

```diff
@@ -3,21 +3,159 @@
+  long batch_delay_millis = static_cast<long>(std::stoi(config.find(CONFIG_KEY_BATCH_DELAY)->second));
+  for (int j = 0; j < num_batches; j++) {
+    std::vector<boost::future<Response>> futures;
+    std::vector<int> binary_labels;
+    for (int i = 0; i < batch_size; i++) {
+      std::srand(time(NULL));
+      int index = std::rand() % 2;
+      std::vector<double> query_vec;
+      if(index == 0) {
+        // Send a plane
+        size_t plane_index = std::rand() % planes_vecs.size();
+        query_vec = planes_vecs[plane_index];
+        binary_labels.emplace_back(SKLEARN_PLANE_LABEL);
+      } else {
+        // Send a bird
+        size_t bird_index = std::rand() % birds_vecs.size();
+        query_vec = birds_vecs[bird_index];
+        binary_labels.emplace_back(SKLEARN_BIRD_LABEL);
+      }
+
+      std::shared_ptr<Input> cifar_input = std::make_shared<DoubleVector>(query_vec);
+      boost::future<Response> future =
+          qp.predict({"test", 0, cifar_input, 100000, "EXP3", {std::make_pair(SKLEARN_MODEL_NAME, 1)}});
+      futures.push_back(std::move(future));
+    }
+
+    std::shared_ptr<std::atomic_int> completed = std::make_shared<std::atomic_int>(0);
+    std::pair<boost::future<void>, std::vector<boost::future<Response>>> results =
+        future::when_all(std::move(futures), completed);
+    results.first.get();
+    for(int i = 0; i < static_cast<int>(results.second.size()); i++) {
+      boost::future<Response>& future = results.second[i];
+      double label = static_cast<double>(binary_labels[i]);
+      double pred = future.get().output_.y_hat_;
+      if(pred == label) {
+        accuracy_ratio->increment(1,1);
+      } else {
+        accuracy_ratio->increment(0,1);
+      }
+    }
+    std::this_thread::sleep_for(std::chrono::milliseconds(batch_delay_millis));
+  }
+}
+
+std::unordered_map<std::string, std::string> setup() {
+  std::unordered_map<std::string, std::string> config;
+  std::string path;
+  std::string num_threads;
+  std::string num_batches;
+  std::string batch_size;
+  std::string batch_delay;
+
+  std::cout << "Before proceeding, run bench/setup_bench.sh from clipper's root directory." << std::endl;
+  std::cout << "Enter a path to the CIFAR10 binary data set: ";
+  std::cin >> path;
+  std::cout << "Enter the number of threads of execution: ";
+  std::cin >> num_threads;
+  std::cout << "Enter the number of request batches to be sent by each thread: ";
+  std::cin >> num_batches;
+  std::cout << "Enter the number of requests per batch: ";
+  std::cin >> batch_size;
+  std::cout << "Enter the delay between batches, in milliseconds: ";
+  std::cin >> batch_delay;
+
+  config.emplace(CONFIG_KEY_PATH, path);
+  config.emplace(CONFIG_KEY_NUM_THREADS, num_threads);
+  config.emplace(CONFIG_KEY_NUM_BATCHES, num_batches);
+  config.emplace(CONFIG_KEY_BATCH_SIZE, batch_size);
+  config.emplace(CONFIG_KEY_BATCH_DELAY, batch_delay);
+  return config;
+};
```

Comment:
Done. We can now specify a path to a JSON configuration file with the command line argument `-f ` or `--filename`. The configuration file should be formatted as follows:

```
{
   "cifar_data_path":"<path to a data_batch_<n>.bin file in the cifar binary data set>",
   "num_threads":"<number of threads>",
   "num_batches":"<number of batches>",
   "batch_size":"<batch size>",
   "batch_delay_millis":"<batch delay millis>"
}
```